docs/gaslessWithERC4337.md

```diff
@@ -0,0 +1,131 @@
+---
+id: gasless-with-ERC4337
+title: How Godwoken experiments gasless transaction using simplified ERC-4337 proposal
+---
+
+This post is supposed to introduce our recent works about bringing a gasless-transaction feature on Godwoken, the generic optimistic roll-up solution built on the CKB network. It covers the basic principle and details about how the gasless feature works based on simplified ERC-4337 specs. We would also love to treat it as an opportunity to discuss the pros and cons of such a model with the most intelligent readers in crypto.
+
+## Basic Idea
+
+Since layer2 is an extended network for the underlying layer1 blockchain, it serves as an ambition to embrace mainstream users, in which most of them are new to the crypto world and hard to onboard, due to the gas problem:
+1. you need to understand how gas works to send a transaction, while most internet services we are using today are free of charge and easy to pay.
+2. you need to have some cryptocurrencies in your wallet before you can pay for the gas to do anything on-chain.
+
+When most users are on their first time onboard, they are often stuck at these two problems at the same time. On the other hand, some developers are eager to gain new users, they are willing to pay gas fees for users to use their product or play their blockchain games. There just doesn't exist an easy way to do that.
+The basic idea of the gasless transaction is to allow users to send free-charged transactions so they can freely use the on-chain application first and then charge back from the application the transaction interacts with. In another way, the application is sponsoring the user's transaction so users don't need to pay for themselves.
+For Godwoken to do that, we need to solve two problems:
+
+## How to let users submit free-charged transactions
+
+The transaction fee in EVM-equivalent blockchains is charged based on the gas price and the total gas used by this transaction. EIP-1559 gives a more complex model about this but since that's out of topic for this post, we will not touch it here.
+
+> gas_fee = gas_price * gas_used
+
+- the gas used is determined by what the transaction is doing (e.g. How much space it writes to the blockchain? Does it consume a lot of CPU to process computation?).
+- the gas price is determined by how much you are willing to bid the block_producer to pack your transaction ASAP.
+
+This mechanism works like a bidding marketplace for transaction fees. The block_producer sorts the transaction by gas price from high to low, then pack the first transaction from the top down in order to make it a profitable business.
+Normally when the gas price is set to 0, the transaction won't be accepted by block_producer, since it takes the cost for the block producer to execute the transaction and included it into a new block.block_producer
```

Comment:
```suggestion
Normally when the gas price is set to 0, the transaction won't be accepted by block_producer, since it takes the cost for the block producer to execute the transaction and included it into a new block.
```